doc/platform_api/CMIS_and_C-CMIS_support_for_ZR.md

```diff
@@ -0,0 +1,1545 @@
+    configure interfaces transceiver (firmware download <bin_file> | firmware commit | firmware run | firmware upgrade <bin_file> | firmware switch) [<interface_name>]
+    ```
+
+- Example (show module firmware version):
+    ```
+    admin@sonic:~$ show interfaces transceiver firmware version Ethernet0 
+    Ethernet0:
+    Get module FW info
+    Image A Version: xxx.xxx.xxx
+    Image B Version: xxx.xxx.xxx
+    Running Image: A; Committed Image: A
+    Get module FW info time: 0.15 s
+    ```
+    
+- Example (configure module firmware download):
+    ```
+    admin@sonic:~$ configure interfaces transceiver firmware download qsfp_dd_ver_xx_xx_xx.bin Ethernet0
+    Start FW downloading
+    Start module FW download: Success
+    Start module FW download time: 0.10 s
+    Total size: 1234567 start bytes: 67 remaining: 1234500
+    Module FW download complete: Success
+    Complete module FW download time: 500.00 s
+    ```
+    
+- Example (configure module firmware run):
+    ```
+    admin@sonic:~$ configure interfaces transceiver firmware run Ethernet0
+    Module FW run: Success
+    Module FW run time: 2.00 s
+    ```
+    
+- Example (configure module firmware commit):
+    ```
+    admin@sonic:~$ configure interfaces transceiver firmware commit Ethernet0
+    Module FW commit: Success
+    Module FW commit time: 2.00 s
+    ```
+
+- Example (configure module firmware upgrade, which combines the show FW version, configure download, show download status, configure commit and run and show version in the end):
+    ```
+    admin@sonic:~$ configure interfaces transceiver firmware upgrade qsfp_dd_ver_xx_xx_xx.bin Ethernet0
+    Ethernet0:
+    Start FW downloading
+    Start module FW download: Success
+    Start module FW download time: 0.10 s
+    Total size: 1234567 start bytes: 67 remaining: 1234500
+    Module FW download complete: Success
+    Complete module FW download time: 500.00 s
+    
+    Module FW run: Success
+    Module FW run time: 2.00 s
+    
+    Module FW commit: Success
+    Module FW commit time: 2.00 s
+  
+    Before switch Image A: xxx.xxx.xxx; Run: 1 Commit: 1, Valid: 0
+    Before switch Image B: xxx.xxx.xxx; Run: 0 Commit: 0, Valid: 0
+    Image A: xxx.xxx.xxx; Run: 0 Commit: 0, Valid: 0
+    Image B: xxx.xxx.xxx; Run: 1 Commit: 1, Valid: 0
+    ```
+
+- Example (configure module firmware switch):
+    ```
+    admin@sonic:~$ configure interfaces transceiver firmware switch Ethernet0
+    Module FW run: Success
+    Module FW run time: 2.00 s
+    
+    Module FW commit: Success
+    Module FW commit time: 2.00 s
+  
+    Before switch Image A: xxx.xxx.xxx; Run: 1 Commit: 1, Valid: 0
+    Before switch Image B: xxx.xxx.xxx; Run: 0 Commit: 0, Valid: 0
+    Image A: xxx.xxx.xxx; Run: 0 Commit: 0, Valid: 0
+    Image B: xxx.xxx.xxx; Run: 1 Commit: 1, Valid: 0
+    ```
+
+### 3. High level functions
+
+#### 3.1 Get module basic information
+- get_module_type
```

Comment:
are you trying to extend the current platform API?

Reply:
CMIS modules have different memory maps. We already complete the work to refactor platform API and include CMIS module support.